### ISSUE TITLE
A11y_VisualStudioCodeClient_Welcome Screen_Contrast:The keyboard visual focus indicator is difficult to see within the status bar.

### ISSUE THREAD

**Issue by kupatkar99** [microsoft/vscode#141709]
### GitHubTags:
#A11yMAS; #A11yTCS; #A11ySev2; #DesktopApp; #Visual Studio Code Client; #BM-VisualStudioCode-Win32-Nov2021; #Win32; #WCAG1.4.11; #NonTextContrast;

### Environment Details:
Application Name: Visual Studio Code Client
Windows Version: Windows 11

### Repro Steps:
1)Turn on Narrtor/NVDA
2) Open Visual studio code editor.
3) Using "Alt+H" key navigate to the "Help" menu bar option .
4) Using "Down" arrow key navigate to the "Welcome" item.
5) Make sure you have the following setting in settings.json: "workbench.startupEditor": "gettingStarted"
6) Using "Alt+H" navigate to "Help" menu bar option and use "down" arrow key to navigate to "Getting started".
7) Verify that whether the keyboard focus indicator is clearly visible in status bar or not.  

### Actual Result:
The keyboard visual focus indicator is difficult to see within the status bar.

### Expected Result:
The keyboard visual focus indicator should be clearly visible within the status bar.

### User Impact:
User will face difficulty in accessing the content if the focus is not clearly visible with in the status bar.
![A11y_VisualStudioCodeClient_Welcome Screen_Contrast](https://user-images.githubusercontent.com/98530069/151531053-ba3980bd-0287-4791-9e52-a0eee337cc22.png)

**Comment by alexdima** [comment]
I don't quite understand what this is about. I think it might be that it is unclear where the focus is after running Help > Get Started

**Comment by isidorn** [comment]
Focus in the status bar when there is a focused item is not high contrast enough.
Let's assign first to @misolori so we try to find a design solution for this.
I just used the hover transparency which is not strong enough. We should improve this

![Screenshot 2022-01-28 at 11 57 08](https://user-images.githubusercontent.com/1926584/151535244-fe187030-c80e-4f59-ac21-52a6ad963e3b.png)

**Comment by alexdima** [comment]
@kupatkar99 Thank you for clarifying with a screenshot.

**Comment by misolori** [comment]
The challenge here is that we use blue for our focus color so we'd need an alternate color used in the status bar that passes the 3:1 contrast ration between the background/border.

**Comment by bpasero** [comment]
We have three colors:
* blue when a folder is opened
* purple when no folder is opened
* orange when debugging

We can do a selective color change only for the status bar, but we have to find a color that can be seen in all three cases. Maybe Devin you can look into which color to use.

On the upside, the status bar color is theme independent.

**Comment by isidorn** [comment]
If we could find a colour that works that would be great. If not we can introduce a focus outline?

**Comment by daviddossett** [comment]
> If we could find a colour that works that would be great. If not we can introduce a focus outline?

I was thinking something along these lines too given the challenge of accounting for different colors across themes. I'll throw some ideas here early this week.

**Comment by misolori** [comment]
I like the simplicity of option 1, option 3 could work but with the closeness to the edge it could be hard to see. 

It's also worth pointing out that a lot of themes don't use a different status bar bg color (the default one) like we do so when we might need to choose the default that works for most themes and we can select a custom color for our themes (we'll do this occasionally):

https://github.com/microsoft/vscode/blob/91f7694e68af5b5a9f05d920a5f0420dcb9c4ff5/extensions/theme-defaults/themes/dark_vs.json#L4-L24

**Comment by daviddossett** [comment]
I was thinking to use `editor.foreground` for the border since that seemed to be the most likely color to always contrast with the background of the status bar in any state.

**Comment by bpasero** [comment]
Did you mean to say `editor.background`? I think foreground is typically black in all themes.

I like the idea of using a different color for the outline that we already have. Maybe we can introduce a new color that by default is white and then existing themes would also be affected and it would only go away if they were to override this color?

**Comment by isidorn** [comment]
I also like Option 1. Especially if this does not increase the overall size of the status bar item - as I am worried if it would look jarring that once the item gets focus it gets larger.
As for the colour - can't we use the existing `statusBar.foreground`, `statusBar.noFolderForeground` and `statusBar.debuggingForeground` depending on the state we are in?

**Comment by daviddossett** [comment]
> Did you mean to say editor.background? I think foreground is typically black in all themes.

I like the idea of using a different color for the outline that we already have. Maybe we can introduce a new color that by default is white and then existing themes would also be affected and it would only go away if they were to override this color?

I may be mistaken—I thought foreground was typically white-ish in most dark themes and black in most light themes. Which wouldn't work on the blue status bar now that I think about it.

I like @isidorn's suggestion of using the statusBar foreground tokens. I'll give that a spin.

**Comment by bpasero** [comment]
I have this running in https://github.com/microsoft/vscode/tree/ben/141709

Unfortunately now every-time you click an entry you see the ugly white border:

![Recording 2022-02-09 at 18 13 49](https://user-images.githubusercontent.com/900690/153253630-13d83f57-435a-4ffd-93e5-a193882474c3.gif)

I forgot how we managed to not show the outline in other places that have it only when keyboard is used.

@daviddossett maybe you want to continue the exploration in that branch.

**Comment by daviddossett** [comment]
Thanks for checking this out! I've also been working on another branch. I'll make sure we don't show the outline on click. I think @Tyriar had the same thoughts on a call earlier.

**Comment by daviddossett** [comment]
Off the top of my head it's something to do with `active` and `focus` overlap but I'll find out and push it up if it works as expected.

**Comment by daviddossett** [comment]
`:focus-visible` seems to be the trick here. 

https://user-images.githubusercontent.com/25163139/153259925-02adb18c-35e2-4294-85db-150418bd786d.mp4

Still need to fix the issue this outline conflicts with the status bar border in some themes.

https://user-images.githubusercontent.com/25163139/153260384-c1842b2c-5d26-4aaf-9b80-f7cb0922685d.mp4

**Comment by daviddossett** [comment]
Here's the relevant commit from [my branch](https://github.com/microsoft/vscode/commit/52559c493b877d87ac55693ca815dc24e8bb29c1). It builds off of Ben's initial work with some updates to the pseudo-class stuff.

**Comment by daviddossett** [comment]
I thought of a potentially interesting solution here by doing some basic conditional styling logic.

Tl;dr ```if (borderColor) { outline-offset: -2px } else { outline-offset: -1px)```

The only downside would be the 1px gap between the base of the status bar and the focus border. Not sure if there's any other cleaner solution?

https://user-images.githubusercontent.com/25163139/153275287-06205734-e569-4b9b-b3a1-2beaa4169638.mp4

**Comment by isidorn** [comment]
Verified this works now -> adding verified label.
Great work!
